mungegithub/mungers/testowner/owner.go

```diff
@@ -107,33 +113,31 @@ func NewReloadingOwnerList(path string) (*ReloadingOwnerList, error) {
 }
 
 // TestOwner returns the owner for a test, or the empty string if none is found.
-func (o *ReloadingOwnerList) TestOwner(testName string) string {
+func (o *HTTPOwnerList) TestOwner(testName string) string {
 	err := o.reload()
 	if err != nil {
-		glog.Errorf("Unable to reload test owners at %s: %v", o.path, err)
+		glog.Errorf("Unable to reload test owners at %s: %v", o.url, err)
 		// Process using the previous data.
 	}
 	return o.ownerList.TestOwner(testName)
 }
 
-func (o *ReloadingOwnerList) reload() error {
-	info, err := os.Stat(o.path)
+func (o *HTTPOwnerList) reload() error {
+	resp, err := o.client.Get(o.url)
```

Comment:
Yes, httpcache respects the HTTP caching RFCs. It also respects the max-age header, so it won't even do an HTTP request to check freshness if the current request is valid.

The munger doesn't open too many new issues, so this is all somewhat premature regardless.